src/compiler/utilities.ts

```diff
@@ -987,7 +987,9 @@ namespace ts {
         return name.kind === SyntaxKind.ComputedPropertyName && !isStringOrNumericLiteralLike(name.expression);
     }
 
-    export function getTextOfPropertyName(name: PropertyName | NoSubstitutionTemplateLiteral): __String {
+    export function getTextOfPropertyName(name: PropertyName | NoSubstitutionTemplateLiteral): __String;
+    export function getTextOfPropertyName(name: PropertyName | NoSubstitutionTemplateLiteral, undefinedIfNotLiteral: boolean): __String | undefined;
+    export function getTextOfPropertyName(name: PropertyName | NoSubstitutionTemplateLiteral, undefinedIfNotLiteral?: boolean): __String | undefined {

```

Comment:
Stylistic nit and maybe others disagree, but I think it’d be a bit cleaner to have `tryGetTextOfPropertName` which can always return `undefined` and `getTextOfPropertyName` which calls the former and asserts a result from it.

Reply:
That's not a bad idea, but I think if I do that, it'll end up not providing any info as to what part of the access went wrong, versus currently, it's asserting.

I'm not sure if it's a pattern to just try/catch/return undefined in this codebase, though, as that would be sort of the best of both worlds.

Reply:
Err, well, probably not the best, since you'd hit the assert when debugging.

Reply:
`getTextOfPropertyName` could preemptively check for non-literal computed properties and throw the same error, but I’m not sure there’s a ton of value in keeping the specific assertion. There’s currently no other way for it to fail and it should be pretty obvious what happened by the time one of us is checking out the source code.

Reply:
Fair enough, I'll do that.

Reply:
It seems like we usually avoid try/catch when possible, probably due to lingering fears from old versions of v8 not being able to optimize functions containing try/catch.